src/main/java/uk/gov/pay/products/service/ProductFinder.java

```diff
@@ -45,6 +45,17 @@ public ProductFinder(ProductDao productDao, LinksDecorator linksDecorator) {
                 .orElseGet(Optional::empty);
     }
 
+    @Transactional
+    public Optional<Product> updatePayApiTokenByExternalId(String externalId, String payApiToken) {
+        return productDao.findByExternalId(externalId)
+                .map(productEntity -> {
+                    productEntity.setPayApiToken(payApiToken);
+                    productDao.merge(productEntity);
```

Comment:
I think the `@Transactional` means you don’t need to explicitly merge but it won’t do any harm.

Reply:
I thought`@Transactional` provides commit and rollback support. It would be nice to test this function against a real product DAO rather than the mocked product DAO.  Using the mocked product DAO in the unit test doesn't give me confidence that it will work without using merge() function.

Reply:
In other words, the unit test doesn't actually test this function using a real mini database.

Reply:
I have tested the `updatePayApiTokenByExternalId` function against a real product DAO. It works fine. When I remove `productDao.merge(productEntity);` from the function, it fails the test. The merge() function is required.